### PR TITLE
Bug fixes to `Remind` feature

### LINKED ISSUE #144: [PE-D] Reminders shown for clients who weren't given any
The above 2 clients did not have reminders at the start but suddenly are also added to the reminder list.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: WJunHong/ped#10

### LINKED ISSUE #171: [PE-D] Remind command does not display tag after completion
Remind command does not display the reminder pop-up till another command is issued

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: aidanaden/ped#10

### LINKED ISSUE #172: [PE-D] Reminder window contains users with no reminders
Reminder window contains a user without any reminders attached to them

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: aidanaden/ped#11

### PR DESCRIPTION
**Pull request by waffledood**
Erroneous behavior of `Remind` feature: 
- Reminder label not displayed in `ReminderWindow` after successful completion of the `RemindCommand`, only displayed after successful completion of `EditCommand` 
    this means the Reminder label will only appear in the `ReminderWindow` when an `EditCommand` is executed on a `Person` who had a `Reminder` previously set for him 

## 
Changes made to resolve these errors: 
- Update the `RemindCommand` to update the `Person` in the `ModelManager` with the latest instance of the `Person` 

Other changes: 
- Update the `hashCode()` of `Person` as `Person` is being used as a `Key` in the `HashMap` in `ReminderPersons`, hence it is necessary to omit the favourite status of a `Person`. 
- Update the `ReminderWindow` to display the "No Reminders Set!" label when there are no `Reminder`s 
- Update the `FavouriteCommand` so that toggling a `Person`'s favourite status will not remove the `Reminder` label 

## 
Updated `Remind` feature as follows: 

https://user-images.githubusercontent.com/59903913/161431417-3257d34f-d463-4771-8047-d7a7b7219f5f.mp4

[Note: This PR has not included the changes made to the behaviour of the Favourite command in #177 ]

## 
This PR closes #172 , closes #144 , closes #171

### CODE REVIEW

File: src/main/java/seedu/address/storage/ReminderPersons.java [hunk 1]
```diff
@@ -33,33 +33,54 @@ public static ReminderPersons getInstance() {
...
+        return reminders.put(person, reminder);
     }
 
     /**
      * Removes a person from the list of persons to be reminded of
      * @param person The person to remove from reminders.
      */
     public Reminder remove(Person person) {
-        return reminderPersons.reminders.remove(person);
+        return reminders.remove(person);
```
Comment (hsiaojietng) [inline src/main/java/seedu/address/storage/ReminderPersons.java:44]: Not really sure what `reminders` here represent, but I am assuming it to be a list that contains the clients to remind. If so, I would suggest to change the name as it does not logically make sense for `reminders` to remove a `Person`. Similarly for the methods below (if this applies).

**Review by hsiaojietng — COMMENTED** [review]
Let me know if I am wrong in the review.

File: src/main/java/seedu/address/storage/ReminderPersons.java (hunk 1 above)
Comment (waffledood) [reply src/main/java/seedu/address/storage/ReminderPersons.java]: got it, adjusted the name accordingly 👍

**Comment by waffledood** [comment]
> Let me know if I am wrong in the review.

not at all, thanks for pointing that out Jiet :) 

it's actually a `HashMap` of `Person`s (key) and the corresponding `Reminder` (value) set, thanks for pointing that out, the original name `variables` made it sound like it was a `List` of `Reminder`s

**Review by hsiaojietng — APPROVED** [review]
Nice, LGTM!

**Pull request merged**